src/core/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,22 @@
+import { Checkbox } from '@material-ui/core';
+import React, { useState } from 'react';
+import CheckOutlinedIcon from '@material-ui/icons/CheckOutlined';
+import { useStyles } from './styles';
+import { CheckboxProps } from '@material-ui/core/Checkbox';
+
+const CheckBox: React.FC<CheckboxProps> = ({ disabled, checked }) => {
+  const classes = useStyles();
+  const [check, setChecked] = useState<boolean | undefined>(checked);
+  return (
+    <Checkbox
+      data-testid="checkbox"
+      className={classes.root}
+      checkedIcon={<CheckOutlinedIcon />}
+      disabled={disabled}
+      checked={check}
+      onChange={() => setChecked(!check)}
```

Comment:
With this change if I pass `Checkbox check={true}` and mutate the state here then checkbox would be unchecked but the place where we have implemented it still would remain `true` which is a loophole

Reply:
@S-ayanide When passing the checkox checked paraemeter=true via props, then checkbox is being checked and when clicking on the checkbox, it becomes unchecked. And vica versa. If there is an issue, let me know pls